File: 414-第三大的数.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        lens = len(nums)
        nums.sort()
        count = 1
        ans = nums[lens-1]
        for i in range(lens-1,-1,-1):
            if ans > nums[i]:
                ans = nums[i]
                count+=1
                if count == 3:
                    return ans;
        return nums[lens-1]

File: 414-第三大的数.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        int lens=nums.size();
        int count=1;
        sort(nums.begin(),nums.end());
        int ans=nums[lens-1];
        for(int i=lens-1;i>=0;i--)
        {
            if(nums[i]<ans)
            {
                ans=nums[i];
                count++;
                if(count==3)
                    return ans;
            }
        }
        return nums[lens-1];
    }
};
